pkg/server/server.go

```diff
@@ -1159,6 +1167,10 @@ func (s *Server) PreStart(ctx context.Context) error {
 		return err
 	}
 
+	// Start loss of quorum recovery server early as it needs to patch
+	// engine data before any replica initialization happens.
+	_ = s.recoveryServer
```

Comment:
Why should it be the recovery server's job to run the recovery? This seems like conflated concerns. Recovery should happen as soon as possible in the startup sequence, i.e. as soon as the engine has been initialized. The recovery server's job is to provide the RPC interface to the `debug recover` commands, and being a networked server, should start much later in the startup sequence.